### PR TITLE
Sorry cypress multiple replicas

### LINKED ISSUE #105: Multiple deployment replicas
- [x] I have searched issues for similar feature requests

## What is this feature about (1 sentence)

Being able to configure more that 1 replica for director, dashboard and API Deployments.

## Why is it needed? What is the value? For whom do we build it?

High Availability. It would prevent issues during pod rescheduling

## Other

Not sure if there is any limitation for Director, Dashboard and API to be highly available?

### PR DESCRIPTION
**Pull request by pkoraca**
Fixes #105

- Allow setting multiple replicas
- Keep replicas defaults to 1
- Use hard node and soft zone anti-affinity
- Use PodDisruptionBudgets if replicas > 1

### CODE REVIEW

**Comment by tico24** [comment]
Can we please at least one simple test in place for this stuff? Feels important.

**Comment by tico24** [comment]
Pop a values file in here with >1 replica on something(s): https://github.com/sorry-cypress/charts/tree/main/charts/sorry-cypress/ci

and then yeh, where appropriate, add a new connection test: https://github.com/sorry-cypress/charts/blob/main/charts/sorry-cypress/templates/test/test-connections.yaml

File: charts/sorry-cypress/values.yaml [hunk 1]
```diff
@@ -162,7 +196,21 @@ director:
...
+            matchLabels:
+              app: {{ include "sorry-cypress-helm.fullname" . }}-director
+          topologyKey: kubernetes.io/hostname
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app: {{ include "sorry-cypress-helm.fullname" . }}-director
+            topologyKey: failure-domain.beta.kubernetes.io/zone
```
Comment (tico24) [inline charts/sorry-cypress/values.yaml:213]: Forgive my ignorance, is this a standardised key for all k8s implementations?

**Comment by pkoraca** [comment]
> Pop a values file in here with >1 replica on something(s): https://github.com/sorry-cypress/charts/tree/main/charts/sorry-cypress/ci
> 
> and then yeh, where appropriate, add a new connection test: https://github.com/sorry-cypress/charts/blob/main/charts/sorry-cypress/templates/test/test-connections.yaml

This will not work. The problem is that now we also have `hostname` anti-affinity that is `required`, and setting replicas > 1 for tests will fail because only 1 pod can be provisioned on a single node `kind` cluster. So to really >1 properly, we would need multi-node cluster to deploy to.

One option would be to change `requiredDuringSchedulingIgnoredDuringExecution` to `preferredDuringSchedulingIgnoredDuringExecution` but then K8S does not guarantee anti-affinity

**Comment by tico24** [comment]
We should be able to add more nodes to the kind cluster. Not something I can do this week, but you're welcome to have a stab at it if you like?

**Comment by pkoraca** [comment]
Looks like it works! Let me know if you'd prefer to have kind config in a different location

**Comment by tico24** [comment]
ah yes, could it either go in the CI directory or the workflows dir (not sure if the latter will cause github to cry?).

Nice stuff. Once that's done, I'll merge it in (probably tomorrow now).

**Pull request merged**